nautobot_ssot/integrations/dna_center/jobs.py

```diff
@@ -105,13 +152,24 @@ def run(
         memory_profiling,
         debug,
         dnac,
+        area_loctype,
+        building_loctype,
+        floor_loctype,
+        location_map,
         bulk_import,
         tenant,
         *args,
         **kwargs,  # pylint: disable=arguments-differ, too-many-arguments
     ):
         """Perform data synchronization."""
         self.dnac = dnac
+        self.area_loctype = area_loctype
+        if not self.area_loctype.nestable:
+            self.logger.error("Area LocationType is not nestable.")
+            raise ConfigurationError("Area LocationType is not nestable.")
```

Comment:
Should we also validate that `floor_loctype.parent == building_loctype` and `building_loctype.parent == area_loctype` here?

Reply:
Probably not a bad idea.

Reply:
I've added a function to do this check before the adapters are loaded.